content/en/blog/_posts/2023-12-13-kubernetes-1.29.md

```diff
@@ -0,0 +1,203 @@
+
+In Kubernetes, volume [access modes](/docs/concepts/storage/persistent-volumes/#access-modes)
+are the way you can define how durable storage is consumed. These access modes are a part of the spec for PersistentVolumes (PVs) and PersistentVolumeClaims (PVCs). When using storage, there are different ways to model how that storage is consumed. For example, a storage system like a network file share can have many users all reading and writing data simultaneously. In other cases maybe everyone is allowed to read data but not write it. For highly sensitive data, maybe only one user is allowed to read and write data but nobody else.
+
+Before v1.22, Kubernetes offered three access modes for PVs and PVCs:
+*	ReadWriteOnce – the volume can be mounted as read-write by a single node
+*	ReadOnlyMany – the volume can be mounted read-only by many nodes
+*	ReadWriteMany – the volume can be mounted as read-write by many nodes
+
+The ReadWriteOnce access mode restricts volume access to a single node, which means it is possible for multiple pods on the same node to read from and write to the same volume. This could potentially be a major problem for some applications, especially if they require at most one writer for data safety guarantees.
+    
+To address this problem, a fourth access mode ReadWriteOncePod was introduced as an Alpha feature in v1.22 for CSI volumes. If you create a pod with a PVC that uses the ReadWriteOncePod access mode, Kubernetes ensures that pod is the only pod across your whole cluster that can read that PVC or write to it. In v1.29, this feature became Generally Available.
+
+### Node volume expansion Secret support for CSI drivers ([SIG Storage](https://github.com/kubernetes/community/tree/master/sig-storage)) {#csi-node-volume-expansion-secrets}
+    
+In Kubernetes, a volume expansion operation may include the expansion of the volume on the node, which involves filesystem resize. Some CSI drivers require secrets, for example a credential for accessing a SAN fabric, during the node expansion for the following use cases:
+	•	When a PersistentVolume represents encrypted block storage, for example using LUKS, you need to provide a passphrase in order to expand the device.
+	•	For various validations, the CSI driver needs to have credentials to communicate with the backend storage system at time of node expansion. 
+
+To meet this requirement, the CSI Node Expand Secret feature was introduced in Kubernetes 1.25. This allows an optional secret field to be sent as part of the NodeExpandVolumeRequest by the CSI drivers so that node volume expansion operation can be performed with the underlying storage system. In Kubernetes 1.29, this feature became generally available.
+
+### Robust data encryption key generation with KMSv2 and KMSv2KDF ([SIG Auth](https://github.com/kubernetes/community/tree/master/sig-auth))
+
```

Comment:
```suggestion

For Kubernetes v1.29, [encryption at rest for API data](/docs/tasks/administer-cluster/encrypt-data/)
can integrate with an external _key management service_ (KMS). Available since Kubernetes v1.13,
this mechanism allows you to protect your cluster's control plane from security risks even if the etcd
persistence layer is somehow compromised.

New in v1.29 is stable support for version 2 of the KMS integration. If you're using KMS for
encryption, or you plan to, the Kubernetes project recommends using the version 2 integration
mechanism in all cases.

For details about integration with specific cloud providers and other services, ask your provider
or consult the documentation for their plugin.

```
?